src/test/hexDocumentModel.test.ts

```diff
@@ -0,0 +1,82 @@
+/*---------------------------------------------------------
+ * Copyright (C) Microsoft Corporation. All rights reserved.
+ *--------------------------------------------------------*/
+
+import { expect } from "chai";
+import { HexDocumentEditOp, HexDocumentModel } from "../../shared/hexDocumentModel";
+import { getTestFileAccessor } from "./util";
+
+describe("HexDocumentModel", () => {
+	let model: HexDocumentModel;
+	beforeEach(async () => {
+		model = new HexDocumentModel({
+			accessor: await getTestFileAccessor(new Uint8Array([0, 1, 2, 3, 4, 5, 6, 7, 8, 9])),
+			supportsLengthChanges: false,
+			isFiniteSize: true,
+		});
+	});
+
+	describe("edit reading", () => {
+		const assertContents = async (value: Uint8Array) => {
+			for (let offset = 0; offset < value.length; offset++) {
+				let built = new Uint8Array();
+				for await (const buf of model.readWithEdits(offset)) {
+					built = Buffer.concat([built, buf]);
+				}
+
+				expect(built).to.deep.equal(Buffer.from(value.subarray(offset)), `expected to be equal at offset ${offset}`);
+			}
+
+			expect(await model.size()).to.equal(value.length);
+		};
+
+		it("reads file verbatim", async () => {
+			await assertContents(new Uint8Array([0, 1, 2, 3, 4, 5, 6, 7, 8, 9]));
+		});
+
+		it("reads with a simple insert", async () => {
+			model.makeEdits([
```

Comment:
Some weird spacing going on in some of these documents. Needs formatting

Reply:
This file is all tabs. But after things are merged I suggest adopting Prettier 🙂